packages/framework/src/Pages/PublicationPage.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Hyde\Pages;
+
+use function file_get_contents;
+use Hyde\Framework\Features\Publications\Models\PublicationType;
+use Hyde\Hyde;
+use Hyde\Markdown\Models\FrontMatter;
+use Hyde\Markdown\Models\Markdown;
+use Illuminate\Support\Facades\Blade;
+use Illuminate\Support\HtmlString;
+use function view;
+
+/**
+ * Publication pages adds an easy way to create custom no-code page types,
+ * with support using a custom front matter schema and Blade templates.
+ *
+ * @see \Hyde\Framework\Testing\Feature\PublicationPageTest
+ */
+class PublicationPage extends Concerns\BaseMarkdownPage
+{
+    public PublicationType $type;
+
+    public static string $sourceDirectory = '';
+    public static string $outputDirectory = '';
+    public static string $template = '__dynamic';
+
+    public static function make(string $identifier = '', FrontMatter|array $matter = [], string|Markdown $markdown = '', ?PublicationType $type = null): static
+    {
+        return new static($type, $identifier, $matter, $markdown);
+    }
+
+    public function __construct(PublicationType $type, string $identifier = '', FrontMatter|array $matter = [], Markdown|string $markdown = '')
```

Comment:
This constructor is breaking, while the type here is required, for interoperability it should be put last. We can then throw an exception later on.

Reply:
Resolved via https://github.com/hydephp/develop/pull/685/commits/f11d84b57bb8f775441e895a114fa9b0e4d39533 which should implicitly throw an exception when trying to assign null to the property.